### ISSUE TITLE
Transfer Pak, Pokémon Stadium

### ISSUE THREAD

**Issue by nijoakim** [libretro/mupen64plus-libretro-nx#435]
Hello,

I'm trying to use the transfer pak feature for Pokémon Stadium and Pokémon Red, but I can not get it to work. I am unsure whether it is due to a bug, my own error or incomplete documentation, but maybe this issue can bring clarity in that.

Retroarch version: 1.10.3 (from git)
Mupen64Plus version: 2.4-Vulkan 9beacb2

If it matters, my OS is Debian bookworm (a.k.a. Debian testing)

MD5s of ROMS used:
Pokemon Stadium (U) [!].z64: ed1378bc12115f71209a77844965ba50
Pokemon Red.gb: 3d45c1ee9abd5738df46d2bdda8b57dc

I use the following save file [Pokemon Red.sav.zip](https://github.com/libretro/mupen64plus-libretro-nx/files/8841080/Pokemon.Red.sav.zip), originally created with VBA-M (libretro) and changed extension from .srm to .sav. (Unzipped of course, but GitHub won't let me upload the unzipped version.)

My procedure is as follows: **(WARNING: This seems to destroy the both the Pokemon Red and Pokemon Stadium save files!)**
- Start Pokemon Stadium
- Go to the "Subsystems" menu
    - Select "Load N64 Transfer Pak" and select "Pokemon Red.sav"
    - Select "Load N64 Transfer Pak" and select "Pokemon Red.gb"
    - Select "Load N64 Transfer Pak" and select "Pokemon Stadium (U) [!].z64"
    - Select "Start N64  Transfer Pak"
    
The game starts, but when the games connected to the transfer paks are showing, an error is displayed, reading: "????? ID ***** Saved file not found". Screenshot: 
![sly](https://user-images.githubusercontent.com/4145914/172076963-5e080d06-4852-44fd-b16e-663204fc6aca.png)

So... Does it look like I'm doing something wrong, or is there a bug?

Cheers,
Joakim

**Comment by nijoakim** [comment]
I investigated this further by examinig the console output when performing the above procedure. Pastebin [here](https://pastebin.com/DvrxmeDp).

The following lines are noteworthy:
```
[INFO] [Environ]: SYSTEM_DIRECTORY: "/home/nijoakim/roms/retroarch/system".
[libretro INFO] GB Loader ROM: /home/nijoakim/roms/retroarch/sly/Pokemon Red.gb - 1048576
[libretro INFO] mupen64plus: GB cart type (13) mbc3 RAM BATT
[libretro WARN] Size mismatch between expected RAM size and effective file size
[libretro INFO] GB Loader RAM: /home/nijoakim/roms/retroarch/sly/Pokemon Red.sav - 32768
```

So it seems Mupen expects the gameboy save filesize to be 32768 bytes. Checking what size it acutally is reveals that it is 4821 bytes, which explains the mismatch. The question is then why the size is wrong. Do different Game Boy emulators create different variations of the save files?

To verify that it's not a recent bug, I also checked it in Mupen commit 3e828021b10c29fd6847f9c199f722da4eaa8803 (just after transfer pak support was added) and the problem persists for that one too.

**Comment by m4xw** [comment]
It might be compressed in RA, cant remember if that was only for savestates.
I will ask another dev when i see them around, pretty sure vba just works
FWIW transfer pak *does* work with stadium, but dont bother trying the GB tower, didnt fix that one yet..
Will close the issue, reopen if deemed required!

**Comment by nijoakim** [comment]
Yes, you were right—Turning off the SaveRAM compression option solved the problem; Doing that, my Game Boy save file aquired the correct size and became playable in Pokemon Stadium!

This should probably be documented somewhere—especially considering the Retroarch's default is having SaveRAM compression on... or better yet, Mupen should probably handle the compressed format also. A suggestion for documentation would be to rewrite the first paragraph directly after the "Using the transfer pack." heading on the documentation for the Mupen core found [here](https://docs.libretro.com/library/mupen64plus/), for example like this:

> First, turn SaveRAM compression off. (Since Mupen is currently not able to decompress Gameboy saves.) You'll find the option under **Saving > SaveRAM compression**. Second, you will have to virtually plug the Transfer Pak, so start a N64 game, then **Quick Menu > Options > Player 1 Pak > transfer** and then close content. Then there are 2 ways of loading: 

Feel free to close again if you deem this is handled. Otherwise, I would keep it open until Mupen handles any format or at least until the issue is documented.

**Comment by m4xw** [comment]
I will think about it

**Comment by PhockYouGaming** [comment]
Currently having the same issue as above. I posted it on Reddit page as well. All above steps are complete, I even turned the SaveRam Compression off, and I'm still showing the above image. 

Things I've done:

1. Checked all files to ensure the hyperextensions are correct.
2. reloading PokeRed and ensured it was readable on GB Emulator
3. Reloaded PokeRed and Pokemon Stadium to ensure files were correct.
4. Consult/Revise all methods prior to reach this spot via Reddit and This page.


Some assistance greatly required, my childhood is hurting.
